### ISSUE TITLE
bug: ion-nav transitions are broken in react 

### ISSUE THREAD

**Issue by sean-perkins** [ionic-team/ionic-framework#25688]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x
- [ ] Nightly

### Current Behavior

Starting in 6.1.15, `IonNav` page transitions are incorrect. 

The changes to support custom sized dialogs, introduced a change to the global stylesheet that is being applied in scenarios where `IonNav` is used within `IonModal`.

Problematic styles:
```css
ion-modal .ion-page {
  position: relative;

  contain: layout style;

  height: 100%;
}
```

### Expected Behavior

`IonNav` transitions should be consistent within a modal or outside of a modal. 

### Steps to Reproduce

1. Clone: https://github.com/milang/ionic-simplevitejs
2. Checkout branch: `recursivemodalnav`
3. `npm install`
4. `npm run dev`
5. Click "Open Modal"
6. Click "Push"
7. Observe the incorrect transition between views

### Code Reproduction URL

https://github.com/milang/ionic-simplevitejs

### Ionic Info

6.1.15+

### Additional Information

Related to: https://github.com/ionic-team/ionic-framework/issues/25677 (caused by the same change, but are effected differently between Angular and React).

**Comment by sean-perkins** [comment]
cc: @milang for visibility.

**Comment by liamdebeasi** [comment]
Here is a dev build if anyone would like test the fix: `6.1.16-dev.11658962095.1f24647d`

**Comment by milang** [comment]
@liamdebeasi @sean-perkins I can confirm that development build `6.1.16-dev.11658962095.1f24647d` works as expected in my environment. Thank you for the quick fix! I assume this fix should eventually be available in Ionic 6.2.1?

**Comment by liamdebeasi** [comment]
Thanks for the issue. This has been resolved via https://github.com/ionic-team/ionic-framework/pull/25689, and a fix will be available in an upcoming release of Ionic Framework.
